File: kotlin/matrix/src/main/kotlin/Matrix.kt
class Matrix(private val matrixAsString: String) {

    val rows = buildRows(matrixAsString);

    fun column(colNr: Int): List<Int> {
        return rows.map { it[colNr-1] }
    }

    fun row(rowNr: Int): List<Int> {
        return rows[rowNr-1]
    }

    fun buildRows(inpString: String): List<List<Int>> {
        return inpString.split('\n').map { it.split(' ').map { it.toInt() } }
    }
}

File: ruby/matrix/matrix.rb
class Matrix
    def initialize(inp)
        @rows = inp.split("\n").map { |row|
            row.split(' ').map { |elem| elem.to_i }
        }
    end

    def row(row_num)
        @rows[row_num-1]
    end

    def column(col_num)
        @rows.map { |row| row[col_num-1] }
    end
end